sdks/evals.do/README.md

```diff
@@ -1 +1,204 @@
+      scoreFunction: async (result, expected) => {
+        // Implementation of the scoring function
+        return { score: 0.85, details: 'Captured 85% of key points' }
+      }
+    },
+    conciseness: {
+      description: 'Measures how concise the summary is while retaining important information',
+      scoreFunction: async (result, expected) => {
+        // Implementation of the scoring function
+        return { score: 0.92, details: 'Excellent brevity' }
+      }
+    }
+  }
+})
+
+// Run the evaluation against a specific AI function
+const results = await summarizationEval.run({
+  target: ai.summarizeArticle,
+  options: {
+    model: 'gpt-4',
+    temperature: 0.2
+  }
+})
+
+console.log(results)
+// {
+//   overallScore: 0.88,
+//   metrics: {
+//     accuracy: { score: 0.85, details: 'Captured 85% of key points' },
+//     conciseness: { score: 0.92, details: 'Excellent brevity' }
+//   },
+//   testCaseResults: [
+//     {
+//       testCase: { input: {...}, expectedOutput: {...} },
+//       output: { summary: '...', keyPoints: [...] },
+//       scores: {
+//         accuracy: { score: 0.85, details: '...' },
+//         conciseness: { score: 0.92, details: '...' }
+//       }
+//     },
+//     // More test case results...
+//   ]
+// }
+```
+
+## Core Concepts
+
+### Evaluations
+
+Evaluations are structured assessments of AI component performance against specific criteria.
+
+```typescript
+import { defineEval } from 'evals.do'
+
+const classificationEval = defineEval({
+  name: 'Sentiment Classification',
+  description: 'Evaluates the accuracy of sentiment classification models',
+  testCases: [...],
+  metrics: {...}
+})
+```
+
+### Metrics
+
+Metrics define the specific aspects of performance being measured.
+
+```typescript
+const metrics = {
+  accuracy: {
+    description: 'Percentage of correct classifications',
+    scoreFunction: async (result, expected) => {
+      const correct = result.label === expected.label
+      return { score: correct ? 1.0 : 0.0, details: correct ? 'Correct' : 'Incorrect' }
+    }
+  },
+  confidence: {
+    description: 'Confidence level of the classification',
+    scoreFunction: async (result, expected) => {
+      // Score based on confidence value when correct, penalty when incorrect
+      const isCorrect = result.label === expected.label
+      const score = isCorrect ? result.confidence : (1 - result.confidence)
+      return { score, details: `Confidence: ${result.confidence}` }
+    }
+  }
+}
+```
+
+### Benchmarks
+
+Compare performance across different models or configurations.
+
+```typescript
+import { createBenchmark } from 'evals.do'
+
+const sentimentBenchmark = createBenchmark({
+  name: 'Sentiment Analysis Benchmark',
+  description: 'Compares different models for sentiment analysis performance',
+  evaluation: sentimentEval,
+  
+  configurations: [
+    { name: 'GPT-4', options: { model: 'gpt-4', temperature: 0.2 } },
```

Comment:
Please check https://lmarena.ai/?leaderboard for whatever the top current models ... these are very old models and represent what was SOTA at your training cutoff date